File: AssetBundleFramework/Assets/Works/Res/Lua/Framework/AudioManager.lua

--UI音频管理类--
AudioManager = {
	IsRootLoaded = false
}
local this = AudioManager

--C# AudioManager单例
local CS_AudioManagerInstance = CS.AudioManager.Instance

--Start--
function AudioManager.InitModule()
	
end

--[[
	
]]
function AudioManager.PlayAudio(audioPath)
	if CS_AudioManagerInstance then
		CS_AudioManagerInstance:PlayAudio(audioPath)
	end
end

--[[
	
]]
function AudioManager.StopAudio()
	if CS_AudioManagerInstance then
		CS_AudioManagerInstance:StopAudio()
	end

end

--[[
	
]]
function AudioManager.PlayAudioAtPoint(audioPath)
	if CS_AudioManagerInstance then
		CS_AudioManagerInstance:PlayAudio(audioPath)
	end
end

File: AssetBundleFramework/Assets/Framework/AudioManager/AudioManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class AudioManager: Singleton<AudioManager>
{
    private Vector3 rootPosition;
    private AudioSource _AudioSource;
    private GameObject _RootObj;
    public void Init(GameObject root)
    {
        root.AddComponent<AudioListener>();
        _AudioSource = root.AddComponent<AudioSource>();
        rootPosition = root.transform.position;
    }
    public void PlayAudio(string audioPath)
    {
        LoadAssetUtility.LoadAudioAsset(audioPath, LoadAudioCallback);
    }
    void LoadAudioCallback(AssetLoaderBase loader, bool state)
    {
        if (state == true)
        {
            AudioLoader audio = (AudioLoader)loader;
            _AudioSource.clip = audio.GetAudioClip();
            _AudioSource.Play();

        }
    }
    public void StopAudio()
    {
        _AudioSource.Stop();
    }

    public void PlayAudioAtPoint(string audioPath,float playTime)
    {
        LoadAssetUtility.LoadAudioAsset(audioPath, LoadAudioAtPointCallback);
    }

   
    void LoadAudioAtPointCallback(AssetLoaderBase loader, bool state)
    {
        if (state == true)
        {
            AudioLoader audio = (AudioLoader)loader;
            AudioSource.PlayClipAtPoint(audio.GetAudioClip(), rootPosition);
        }
    }
}
